### ISSUE TITLE
no matching function for call to ‘std::__cxx11::basic_string<char>::basic_string(signed char [1], int&)’

### ISSUE THREAD

**Issue by d-torrance** [Macaulay2/M2#1673]
I just got the following on a build of a Debian package based on the `development` branch in arm64 using Linaro Developer Cloud.  I'm guessing this may have something to do with f67b482?

```
g++ -pipe -Wall -Wno-shadow -Wcast-qual -Wno-sign-conversion -Wno-sign-compare -Wno-parentheses -Wno-sign-compare  -fno-stack-protector -Wuninitialized -I/usr/include  -Wno-error=pragmas -std=gnu++14 -g -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -g3 -O2  -Wno-mismatched-tags -Wno-unused-local-typedefs -Wno-unknown-pragmas -Wno-mismatched-tags -I. -DNDEBUG -I/<<PKGBUILDDIR>>/M2/include -I/<<PKGBUILDDIR>>/M2/include -I/<<PKGBUILDDIR>>/M2/usr-host/include -I/<<PKGBUILDDIR>>/M2/usr-host/include/cdd  -I/<<PKGBUILDDIR>>/M2/include -I/<<PKGBUILDDIR>>/M2/include -I/<<PKGBUILDDIR>>/M2/usr-host/include -I/<<PKGBUILDDIR>>/M2/usr-host/include/cdd  -I/<<PKGBUILDDIR>>/M2/include -I/<<PKGBUILDDIR>>/M2/include -I/<<PKGBUILDDIR>>/M2/usr-host/include -I/<<PKGBUILDDIR>>/M2/usr-host/include/cdd  -Wdate-time -D_FORTIFY_SOURCE=2 -isystem /usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/aarch64-linux-gnu/singular -I/usr/include/singular -I/usr/include/aarch64-linux-gnu/singular -I/usr/include/singular -I/usr/include/aarch64-linux-gnu/singular -I/usr/include/singular   -I/usr/src/gtest  -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include -I/usr/local/include -DNDEBUG  -I/usr/include/cdd -I/usr/include/eigen3 -I/usr/include -DBOOST_STACKTRACE_LINK -isystem /usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/aarch64-linux-gnu/singular -I/usr/include/singular -I/usr/include/aarch64-linux-gnu/singular -I/usr/include/singular -I/usr/include/aarch64-linux-gnu/singular -I/usr/include/singular   -I/usr/src/gtest  -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include -I/usr/local/include -DNDEBUG  -I/usr/include/cdd -I/usr/include/eigen3 -I/usr/include -DBOOST_STACKTRACE_LINK -isystem /usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/aarch64-linux-gnu/singular -I/usr/include/singular -I/usr/include/aarch64-linux-gnu/singular -I/usr/include/singular -I/usr/include/aarch64-linux-gnu/singular -I/usr/include/singular   -I/usr/src/gtest  -Wdate-time -D_FORTIFY_SOURCE=2 -I/usr/include -I/usr/local/include -DNDEBUG  -I/usr/include/cdd -I/usr/include/eigen3 -I/usr/include -DBOOST_STACKTRACE_LINK -I../d -I./../c -Wno-fatal-errors  -c -Wno-cast-qual util.cpp -o util.o
util.cpp: In function ‘void M2_ArrayString_to_stdvector(M2_ArrayString, std::vector<std::__cxx11::basic_string<char> >&)’:
util.cpp:13:37: error: no matching function for call to ‘std::__cxx11::basic_string<char>::basic_string(signed char [1], int&)’
   13 |       std::string b(a->array, a->len);
      |                                     ^
In file included from /usr/include/c++/10/string:55,
                 from util.cpp:5:
/usr/include/c++/10/bits/basic_string.h:625:9: note: candidate: ‘template<class _InputIterator, class> std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&) [with _InputIterator = _InputIterator; <template-parameter-2-2> = <template-parameter-1-2>; _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  625 |         basic_string(_InputIterator __beg, _InputIterator __end,
      |         ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:625:9: note:   template argument deduction/substitution failed:
util.cpp:13:37: note:   deduced conflicting types for parameter ‘_InputIterator’ (‘signed char*’ and ‘int’)
   13 |       std::string b(a->array, a->len);
      |                                     ^
In file included from /usr/include/c++/10/string:55,
                 from util.cpp:5:
/usr/include/c++/10/bits/basic_string.h:587:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&&, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  587 |       basic_string(basic_string&& __str, const _Alloc& __a)
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:587:35: note:   no known conversion for argument 1 from ‘signed char [1]’ to ‘std::__cxx11::basic_string<char>&&’
  587 |       basic_string(basic_string&& __str, const _Alloc& __a)
      |                    ~~~~~~~~~~~~~~~^~~~~
/usr/include/c++/10/bits/basic_string.h:583:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  583 |       basic_string(const basic_string& __str, const _Alloc& __a)
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:583:40: note:   no known conversion for argument 1 from ‘signed char [1]’ to ‘const std::__cxx11::basic_string<char>&’
  583 |       basic_string(const basic_string& __str, const _Alloc& __a)
      |                    ~~~~~~~~~~~~~~~~~~~~^~~~~
/usr/include/c++/10/bits/basic_string.h:579:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::initializer_list<_Tp>, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  579 |       basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:579:45: note:   no known conversion for argument 1 from ‘signed char [1]’ to ‘std::initializer_list<char>’
  579 |       basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~^~~
/usr/include/c++/10/bits/basic_string.h:552:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  552 |       basic_string(basic_string&& __str) noexcept
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:552:7: note:   candidate expects 1 argument, 2 provided
/usr/include/c++/10/bits/basic_string.h:540:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type, _CharT, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’ (near match)
  540 |       basic_string(size_type __n, _CharT __c, const _Alloc& __a = _Alloc())
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:540:7: note:   conversion of argument 1 would be ill-formed:
util.cpp:13:24: error: invalid conversion from ‘signed char*’ to ‘std::__cxx11::basic_string<char>::size_type’ {aka ‘long unsigned int’} [-fpermissive]
   13 |       std::string b(a->array, a->len);
      |                     ~~~^~~~~
      |                        |
      |                        signed char*
In file included from /usr/include/c++/10/string:55,
                 from util.cpp:5:
/usr/include/c++/10/bits/basic_string.h:525:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  525 |       basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:525:53: note:   no known conversion for argument 2 from ‘int’ to ‘const std::allocator<char>&’
  525 |       basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
      |                                       ~~~~~~~~~~~~~~^~~~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:510:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’ (near match)
  510 |       basic_string(const _CharT* __s, size_type __n,
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:510:7: note:   conversion of argument 1 would be ill-formed:
util.cpp:13:24: error: invalid conversion from ‘signed char*’ to ‘const char*’ [-fpermissive]
   13 |       std::string b(a->array, a->len);
      |                     ~~~^~~~~
      |                        |
      |                        signed char*
In file included from /usr/include/c++/10/string:55,
                 from util.cpp:5:
/usr/include/c++/10/bits/basic_string.h:492:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
  492 |       basic_string(const basic_string& __str, size_type __pos,
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:492:7: note:   candidate expects 4 arguments, 2 provided
/usr/include/c++/10/bits/basic_string.h:476:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
  476 |       basic_string(const basic_string& __str, size_type __pos,
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:476:7: note:   candidate expects 3 arguments, 2 provided
/usr/include/c++/10/bits/basic_string.h:461:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
  461 |       basic_string(const basic_string& __str, size_type __pos,
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:461:40: note:   no known conversion for argument 1 from ‘signed char [1]’ to ‘const std::__cxx11::basic_string<char>&’
  461 |       basic_string(const basic_string& __str, size_type __pos,
      |                    ~~~~~~~~~~~~~~~~~~~~^~~~~
/usr/include/c++/10/bits/basic_string.h:448:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  448 |       basic_string(const basic_string& __str)
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:448:7: note:   candidate expects 1 argument, 2 provided
/usr/include/c++/10/bits/basic_string.h:440:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  440 |       basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:440:7: note:   candidate expects 1 argument, 2 provided
/usr/include/c++/10/bits/basic_string.h:431:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string() [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  431 |       basic_string()
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:431:7: note:   candidate expects 0 arguments, 2 provided
In file included from /usr/include/aarch64-linux-gnu/c++/10/bits/c++allocator.h:33,
                 from /usr/include/c++/10/bits/allocator.h:46,
                 from /usr/include/c++/10/memory:64,
                 from stdinc-m2.hpp:7,
                 from util.cpp:3:
/usr/include/c++/10/ext/new_allocator.h: In instantiation of ‘void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = std::__cxx11::basic_string<char>; _Args = {signed char (&)[1], int&}; _Tp = std::__cxx11::basic_string<char>]’:
/usr/include/c++/10/bits/alloc_traits.h:512:17:   required from ‘static void std::allocator_traits<std::allocator<_Tp1> >::construct(std::allocator_traits<std::allocator<_Tp1> >::allocator_type&, _Up*, _Args&& ...) [with _Up = std::__cxx11::basic_string<char>; _Args = {signed char (&)[1], int&}; _Tp = std::__cxx11::basic_string<char>; std::allocator_traits<std::allocator<_Tp1> >::allocator_type = std::allocator<std::__cxx11::basic_string<char> >]’
/usr/include/c++/10/bits/vector.tcc:115:30:   required from ‘void std::vector<_Tp, _Alloc>::emplace_back(_Args&& ...) [with _Args = {signed char (&)[1], int&}; _Tp = std::__cxx11::basic_string<char>; _Alloc = std::allocator<std::__cxx11::basic_string<char> >]’
util.cpp:24:43:   required from here
/usr/include/c++/10/ext/new_allocator.h:150:4: error: no matching function for call to ‘std::__cxx11::basic_string<char>::basic_string(signed char [1], int&)’
  150 |  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/10/string:55,
                 from util.cpp:5:
/usr/include/c++/10/bits/basic_string.h:625:9: note: candidate: ‘template<class _InputIterator, class> std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(_InputIterator, _InputIterator, const _Alloc&) [with _InputIterator = _InputIterator; <template-parameter-2-2> = <template-parameter-1-2>; _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  625 |         basic_string(_InputIterator __beg, _InputIterator __end,
      |         ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:625:9: note:   template argument deduction/substitution failed:
In file included from /usr/include/aarch64-linux-gnu/c++/10/bits/c++allocator.h:33,
                 from /usr/include/c++/10/bits/allocator.h:46,
                 from /usr/include/c++/10/memory:64,
                 from stdinc-m2.hpp:7,
                 from util.cpp:3:
/usr/include/c++/10/ext/new_allocator.h:150:4: note:   deduced conflicting types for parameter ‘_InputIterator’ (‘signed char*’ and ‘int’)
  150 |  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/10/string:55,
                 from util.cpp:5:
/usr/include/c++/10/bits/basic_string.h:587:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&&, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  587 |       basic_string(basic_string&& __str, const _Alloc& __a)
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:587:35: note:   no known conversion for argument 1 from ‘signed char [1]’ to ‘std::__cxx11::basic_string<char>&&’
  587 |       basic_string(basic_string&& __str, const _Alloc& __a)
      |                    ~~~~~~~~~~~~~~~^~~~~
/usr/include/c++/10/bits/basic_string.h:583:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  583 |       basic_string(const basic_string& __str, const _Alloc& __a)
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:583:40: note:   no known conversion for argument 1 from ‘signed char [1]’ to ‘const std::__cxx11::basic_string<char>&’
  583 |       basic_string(const basic_string& __str, const _Alloc& __a)
      |                    ~~~~~~~~~~~~~~~~~~~~^~~~~
/usr/include/c++/10/bits/basic_string.h:579:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::initializer_list<_Tp>, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  579 |       basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:579:45: note:   no known conversion for argument 1 from ‘signed char [1]’ to ‘std::initializer_list<char>’
  579 |       basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~^~~
/usr/include/c++/10/bits/basic_string.h:552:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  552 |       basic_string(basic_string&& __str) noexcept
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:552:7: note:   candidate expects 1 argument, 2 provided
/usr/include/c++/10/bits/basic_string.h:540:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type, _CharT, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’ (near match)
  540 |       basic_string(size_type __n, _CharT __c, const _Alloc& __a = _Alloc())
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:540:7: note:   conversion of argument 1 would be ill-formed:
In file included from /usr/include/aarch64-linux-gnu/c++/10/bits/c++allocator.h:33,
                 from /usr/include/c++/10/bits/allocator.h:46,
                 from /usr/include/c++/10/memory:64,
                 from stdinc-m2.hpp:7,
                 from util.cpp:3:
/usr/include/c++/10/ext/new_allocator.h:150:4: error: invalid conversion from ‘signed char*’ to ‘std::__cxx11::basic_string<char>::size_type’ {aka ‘long unsigned int’} [-fpermissive]
  150 |  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |    |
      |    signed char*
In file included from /usr/include/c++/10/string:55,
                 from util.cpp:5:
/usr/include/c++/10/bits/basic_string.h:525:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  525 |       basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:525:53: note:   no known conversion for argument 2 from ‘int’ to ‘const std::allocator<char>&’
  525 |       basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
      |                                       ~~~~~~~~~~~~~~^~~~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:510:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’ (near match)
  510 |       basic_string(const _CharT* __s, size_type __n,
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:510:7: note:   conversion of argument 1 would be ill-formed:
In file included from /usr/include/aarch64-linux-gnu/c++/10/bits/c++allocator.h:33,
                 from /usr/include/c++/10/bits/allocator.h:46,
                 from /usr/include/c++/10/memory:64,
                 from stdinc-m2.hpp:7,
                 from util.cpp:3:
/usr/include/c++/10/ext/new_allocator.h:150:4: error: invalid conversion from ‘signed char*’ to ‘const char*’ [-fpermissive]
  150 |  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |    |
      |    signed char*
In file included from /usr/include/c++/10/string:55,
                 from util.cpp:5:
/usr/include/c++/10/bits/basic_string.h:492:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
  492 |       basic_string(const basic_string& __str, size_type __pos,
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:492:7: note:   candidate expects 4 arguments, 2 provided
/usr/include/c++/10/bits/basic_string.h:476:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
  476 |       basic_string(const basic_string& __str, size_type __pos,
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:476:7: note:   candidate expects 3 arguments, 2 provided
/usr/include/c++/10/bits/basic_string.h:461:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&, std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
  461 |       basic_string(const basic_string& __str, size_type __pos,
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:461:40: note:   no known conversion for argument 1 from ‘signed char [1]’ to ‘const std::__cxx11::basic_string<char>&’
  461 |       basic_string(const basic_string& __str, size_type __pos,
      |                    ~~~~~~~~~~~~~~~~~~~~^~~~~
/usr/include/c++/10/bits/basic_string.h:448:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const std::__cxx11::basic_string<_CharT, _Traits, _Alloc>&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  448 |       basic_string(const basic_string& __str)
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:448:7: note:   candidate expects 1 argument, 2 provided
/usr/include/c++/10/bits/basic_string.h:440:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _Alloc&) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  440 |       basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:440:7: note:   candidate expects 1 argument, 2 provided
/usr/include/c++/10/bits/basic_string.h:431:7: note: candidate: ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::basic_string() [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>]’
  431 |       basic_string()
      |       ^~~~~~~~~~~~
/usr/include/c++/10/bits/basic_string.h:431:7: note:   candidate expects 0 arguments, 2 provided
make[4]: *** [Makefile.common:33: util.o] Error 1
make[4]: Leaving directory '/<<PKGBUILDDIR>>/M2/Macaulay2/e'
make[3]: *** [Makefile:14: all-in-e] Error 2
make[3]: Leaving directory '/<<PKGBUILDDIR>>/M2/Macaulay2'
make[2]: *** [GNUmakefile:229: all-in-Macaulay2] Error 2
make[2]: Leaving directory '/<<PKGBUILDDIR>>/M2'
```

**Comment by d-torrance** [comment]
I suppose we could have a helper function that converts an `M2_string` to a `std::string` that iterates through each element and casts it to `char`.

**Comment by d-torrance** [comment]
I just discovered `reinterpret_cast`, which should be much simpler.  I'll try that out.

**Comment by DanGrayson** [comment]
I think it's odd that "signed char" and "char" are different types.

**Comment by DanGrayson** [comment]
Should be fixed in commit be70ed263b4f871893a1d00942bcfd5644e88270

**Comment by d-torrance** [comment]
> Should be fixed in commit [be70ed2](https://github.com/Macaulay2/M2/commit/be70ed263b4f871893a1d00942bcfd5644e88270)

Pretty sure that did it!  I'm getting another (I believe unrelated) build error on arm64 now, so I'll leave this issue open until I'm certain that it builds.

**Comment by d-torrance** [comment]
Everything's building nicely on armhf now (#1674 seems to be arm64-specific), so closing.  Thanks!
